src/structurizr/api/structurizr_client.py

```diff
@@ -192,25 +228,41 @@ def put_workspace(self, workspace: Workspace) -> None:
                 f"Error message: {body.get('message', '')}"
             )
 
-    def lock_workspace(self) -> bool:
+    def _lock_workspace(self) -> Tuple[bool, bool]:
         """
         Lock the Structurizr workspace.
 
         Returns:
-            bool: `True` if the workspace could be locked, `False` otherwise.
-
+            bool: `True` if lock succeeeded
+            bool: `True` if on paid plan
         """
         request = self._client.build_request("PUT", self._lock_url, params=self._params)
         request.headers.update(self._add_headers(request))
         response = self._client.send(request)
-        logger.debug("%r", response.json())
         response.raise_for_status()
+        logger.debug("%r", response.json())
         response = APIResponse.parse_raw(response.text)
         if not response.success:
-            logger.error(
+            logger.warning(
                 f"Failed to lock workspace {self.workspace_id}. {response.message}"
             )
-        return response.success
+        return response.success, self._paid_plan(response)
+
+    def lock_workspace(self) -> bool:
+        """Lock the Structurizr workspace.
+
+        Returns:
+            bool: `True` if the workspace could be locked, `False` otherwise.
+
+        Note that free plan Structurizr licences do not support locking.  By
+        default this failure will be ignored, however if you do want such locks
+        to fail then set ignore_free_plan_locking_errors to False.
+        """
+        success, paid_plan = self._lock_workspace()
+        if not success and not paid_plan and self.ignore_free_plan_locking_errors:
+            logger.debug("Ignoring lock failure on free plan")
```

Comment:
I wonder if this is better logged as a warning saying that you cannot lock on the free plan and remove the `ignore_free_plan_locking_errors` attribute? I don't see anyone ever setting that attribute to `False`. (I might be wrong, of course.)

Reply:
Yeah, I was in two minds about that.  I'll remove the attribute so it doesn't confuse.